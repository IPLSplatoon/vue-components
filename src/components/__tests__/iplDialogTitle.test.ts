import IplPanelTitle from '../iplDialogTitle.vue';
import { config, mount } from '@vue/test-utils';

describe('IplDialogTitle', () => {
    config.global.stubs = {
        FontAwesomeIcon: true
    };

    it('matches snapshot', () => {
        const wrapper = mount(IplPanelTitle, { props: { title: 'Cool Dialog' } });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot when closing is disabled', () => {
        const wrapper = mount(IplPanelTitle, { props: { title: 'Cool Dialog', closingDisabled: true } });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('emits event on close button click', async () => {
        const wrapper = mount(IplPanelTitle, { props: { title: 'Cool Dialog' } });

        await wrapper.get('[data-test="close-button"]').trigger('click');

        expect(wrapper.emitted('close')?.length).toEqual(1);
    });
});
