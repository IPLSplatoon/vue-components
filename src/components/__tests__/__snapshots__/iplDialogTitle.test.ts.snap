// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplDialogTitle matches snapshot 1`] = `
"<div class="ipl-dialog-title color-primary"><span class="ipl-dialog-title__title-text">Cool Dialog</span>
  <ipl-button-stub icon="[object Object]" color="transparent" disabled="false" async="false" progressmessage="Working..." successmessage="Done!" errormessage="Error!" requiresconfirmation="false" shortconfirmationmessage="false" small="true" disableonsuccess="false" inline="true" data-test="close-button" class="close-button"></ipl-button-stub>
</div>"
`;

exports[`IplDialogTitle matches snapshot when closing is disabled 1`] = `
"<div class="ipl-dialog-title color-primary"><span class="ipl-dialog-title__title-text">Cool Dialog</span>
  <ipl-button-stub icon="[object Object]" color="transparent" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" errormessage="Error!" requiresconfirmation="false" shortconfirmationmessage="false" small="true" disableonsuccess="false" inline="true" data-test="close-button" class="close-button"></ipl-button-stub>
</div>"
`;
